proc_scripts/correlation_alignment.py

```diff
@@ -0,0 +1,149 @@
+        s_copy = s.C
+        freq_filter = True
+        s_copy.ft('t2')
+        sigma = sigma
+        s_copy *= exp(-(s_copy.fromaxis('t2')-nu_center)**2/(2*sigma**2))
+        s_copy.ift('t2')
+        s_copy2 = s.C
+        s_copy *= nddata(r_[1.,1.,1.,1.],'DeltaPh2')
+        s_copy *= nddata(r_[1.,1.],'DeltaPh1')
+        s_copy.setaxis('DeltaPh2','#')
+        s_copy.setaxis('DeltaPh1','#')
+        correl = s_copy * 0 
+        for ph1_index in range(ph1_len):
+            s_copy['DeltaPh1',ph1_index] = s_copy['DeltaPh1',ph1_index].run(lambda x, 
+                axis=None: roll(x, ph1_index,axis=axis),'ph1')
+        for ph2_index in range(ph2_len):
+            s_copy['DeltaPh2',ph2_index] = s_copy['DeltaPh2',ph2_index].run(lambda x,
+                    axis=None: roll(x,ph2_index,axis=axis),'ph2')
+        for j in range(1,N):
+            correl += s_copy2 * s_copy.C.run(lambda x, axis=None: roll(x,j,axis=axis),
+                indirect_dim).run(conj)
+        if my_iter == 0:
+            logger.info(strm("holder"))
+            if fl is not None:
+                fl.next('Look at correlation function - time domain')
+                fl.image(correl.C.setaxis('vd','#').set_units('vd','scan #'))
+        if my_iter ==0:
+            logger.info(strm("holder"))
+            if fl is not None:
+                fl.next('correlation function\ntime domain, after apod')
+                fl.image(correl.C.setaxis('vd','#').set_units('vd','scan #'))
+        correl.ft_clear_startpoints('t2')
+        correl.ft('t2', shift=True, pad=2**14)
+        correl.ft(['DeltaPh1','DeltaPh2'])
+        correl = correl['DeltaPh1',ph1_selection]['DeltaPh2',ph2_selection] + correl['DeltaPh1',0]['DeltaPh2',0]
+        if my_iter ==0:
+            logger.info(strm("holder"))
+            if fl is not None:
+                fl.next('correlation function \nfreq domain, after apod')
+                fl.image(correl.C.setaxis('vd','#').set_units('vd','scan #'))
+        f_shift = correl.run(real).argmax('t2')
+        #f_shift = nddata(f_shift.data,[indirect_dim])
+        s_copy = s.C
+        s_copy *= exp(-1j*2*pi*f_shift*s_copy.fromaxis('t2'))
+        s.ft('t2')
+        s_copy.ft('t2')
+        if my_iter ==0:
+            logger.info(strm("holder"))
+            if fl is not None:
+                fl.next('after correlation\nbefore ph0 restore')
+                fl.image(s_copy.C.setaxis('vd','#').set_units('vd','scan #'))
+        logger.info(strm('signal energy per transient (recalc to check that it stays the same):',(abs(s_copy**2).data.sum().item() / N)))
+
+        this_E = (abs(s_copy.C.sum(indirect_dim))**2).data.sum().item() / N**2
+        energy_vals.append(this_E / sig_energy)
+        logger.info(strm('averaged signal energy (per transient):', this_E))
+        if last_E is not None:
+            energy_diff = (this_E - last_E)/sig_energy
+            logger.info(strm(energy_diff))
+            if abs(energy_diff) < tol:
```

Comment:
also require that at least 4 iterations have been run

Reply:
not quite -- I added a line to prevent the break if a minimum number of iterations hasn't been reached